library/ssl_tls.c

```diff
@@ -1073,6 +1081,46 @@ static int ssl_tls12_populate_transform( mbedtls_ssl_transform *transform,
         goto end;
     }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    if( ( status = mbedtls_cipher_to_psa( cipher_info->type,
+                                 transform->taglen,
+                                 &alg,
+                                 &key_type,
```

Comment:
@mpg  Shouldn't we store `key_type` in transform struct the same way like `psa_alg`? 

`PSA_BLOCK_CIPHER_BLOCK_LENGTH(psa_key_type_t)` requires key type and it seems it is not stored.

Reply:
I haven't looked at this PR specifically, but yes, it does seem likely to me that the transform needs to store both the algorithm and the key type.

Reply:
It seems to me the key type can be retrieved from the key itself (which is stored in the transform).
```
    psa_key_attributes_t attr = PSA_KEY_ATTRIBUTES_INIT;
    (void) psa_get_key_attributes( key, &attr );
    psa_key_type_t key_type = psa_get_key_type( &attr );
```
Also, where do we need to know the key type outside of the two `populate_tranform` functions, which already have a local variable for it? (Same for key length: I think it can be retrieved from the key if and when we need it, but I'm also not convinced we need it outside `populate_transform` functions.)

Reply:
Och ok. I forgot to set key type in key attributes. 

I need this for `PSA_BLOCK_CIPHER_BLOCK_LENGTH(psa_key_type_t)` in `mbedtls_ssl_get_record_expansion()`.

Reply:
> I need this for `PSA_BLOCK_CIPHER_BLOCK_LENGTH(psa_key_type_t)` in `mbedtls_ssl_get_record_expansion()`.

Oh, good point, I forgot about that one.